### ISSUE TITLE
rolling upgrades seem to have stopped working

### ISSUE THREAD

**Issue by jstrachan** [fabric8io/fabric8-pipeline-library#45]
the old version is scaled down too quickly before the new version is ready. Not sure why but it seems a regression

**Comment by rawlingsj** [comment]
Can you remember if this for Java projects?  

I switched the non java pipelines to deployments / deployment configs recently and noticed the rolling updates working nicely so I think its just Java pipelines we need to look at.

**Comment by jstrachan** [comment]
Last time I looked the rolling upgrades were broken for java projects; though its odd; they should all use the `kubernetesApply` stuff right? I did wonder if its changes to the Controller / kubernetes-client in how it does updates to the `Deployment` / `DeploymentConfig`.

Are the non-java projects using a Deployment and DeploymentConfig now?

**Comment by rawlingsj** [comment]
> Last time I looked the rolling upgrades were broken for java projects; though its odd; they should all use the kubernetesApply stuff right? 

yeah that's right - ok cool that narrows it down so just needs a bit of digging around the controller and client perhaps

> Are the non-java projects using a Deployment and DeploymentConfig now?

yeah that's all done now however I'd like to commit them back to the repo so folks can customise them easily but thats another thing.  I've just raise https://github.com/fabric8io/fabric8-pipeline-library/issues/75 to do this.
